lib/xilinx/configuration.cc

```diff
@@ -41,6 +41,31 @@ Configuration<Series7>::createType2ConfigurationPacketData(
 	return packet_data;
 }
 
+template <>
+Configuration<UltraScale>::PacketData
+Configuration<UltraScale>::createType2ConfigurationPacketData(
+    const Frames<UltraScale>::Frames2Data& frames,
+    absl::optional<UltraScale::Part>& part) {
+	PacketData packet_data;
+	for (auto& frame : frames) {
+		std::copy(frame.second.begin(), frame.second.end(),
+		          std::back_inserter(packet_data));
+
+		auto next_address = part->GetNextFrameAddress(frame.first);
+		if (next_address &&
+		    (next_address->block_type() != frame.first.block_type() ||
+		     next_address->is_bottom_half_rows() !=
+		         frame.first.is_bottom_half_rows() ||
+		     next_address->row() != frame.first.row())) {
+			packet_data.insert(packet_data.end(),
+			                   UltraScale::words_per_frame * 2, 0);
```

Comment:
Please make a constant of the `2`.  Is it just `sizeof(uint16_t)`, etc?

Reply:
The `2` here has nothing to do with the word size. It's has just been empirically proven that at the end of the Type 2 packet there are 2 full (2 * ArchType::words_per_frame) frames with zeros. Adding the `sizeof(uint16_t)` would be misleading as it would imply that the number of zero frames has something to do with the data type which is not the case.

Reply:
Ok sure, so put the details you just wrote into a comment above a new constant with a good name.